flink-runtime/src/main/java/org/apache/flink/runtime/minicluster/MiniCluster.java

```diff
@@ -279,6 +282,10 @@ public void start() throws Exception {
                     miniClusterConfiguration.getRpcServiceSharing() == RpcServiceSharing.SHARED;
 
             try {
+                final PluginManager pluginManager =
+                        PluginUtils.createPluginManagerFromRootFolder(configuration);
+                FileSystem.initialize(configuration, pluginManager);
```

Comment:
This seems sketchy. Because `FileSystem` is partially a singleton this could have surprising effects on other tests if a FS is provided for one of the default schemes (e.g., a custom LocalFileSystem).

Reply:
Would it be safer to move it to the `LocalExecutor`?

Reply:
I don't think so because all tests using the `MiniClusters` implicitly go through the `LocalEnvironment.`

While it is true that users cannot set credentials via the configuration passed to the environment, they could call `FileSystem#initialize()` themselves.
Yes, it is surprising and not ergonomic and all that.

Alternatively we would need a way to reset the `FileSystem` (i.e., call initialize again with no configuration) after each test.

Reply:
How about we add a test-only `uninitialize`? Wouldn't that be the cleanest option?

Reply:
And who calls that? Every test that uses a filesystem? Every test that does something special? How do you enforce that?

Reply:
Every test that calls `initialize` must call `uninitialize`. We can't enforce it but rather make sure that all existing tests are abiding it.

Reply:
Tests might not explicitly call initialize; if they have a factory on the CP then it is loaded implicitly on first use.